tests/snapshots.rs

```diff
@@ -198,12 +200,13 @@ fn check_output_glsl(
 }
 
 #[cfg(feature = "hlsl-out")]
-fn check_output_hlsl(module: &naga::Module, destination: &PathBuf) {
+fn check_output_hlsl(module: &naga::Module, destination: &PathBuf, stage: naga::ShaderStage) {
```

Comment:
why are we having the stage here? `hlsl::write_string` doesn't depend on it, so neither should the test output file.

Reply:
Ah yes right, well that is due to the naming for the snapshots. The dxc compiler has an argument (can't remember exactly what it was) that is required to specify the profile. So that will be shadermodel 6.0 (or higher) and the stage of the shader. The entrypoint also has to be specified, but for now I think we can assume that all test shaders their entrypoint names will be "name".

Reply:
I wasn't that far yet, but take a look at the arguments that [Shader Playground](https://github.com/tgjones/shader-playground/blob/master/src/ShaderPlayground.Core/Compilers/Dxc/DxcCompiler.cs#L142) passes to the dxc compiler

Reply:
Interesting. We need one of the following:
  1. Find/request a way from DXC to check all the entry points - https://github.com/microsoft/DirectXShaderCompiler/issues/3714
  2. Generate a small file per test case with a list of entry points and their execution models, so that we can iterate the lines and feed them into DXC